instructure/canvas-ios
# Offline sync retry logic

refs: MBL-16885
affects: Student
release note: none
test: 
1. I suggest to use my account because "Test" course has 100+ MB of files so you can play around with the retry logic. 
2. Start a course sync 
3. Open the progress screen
4. Turn off Wi-Fi
5. Turn back the Wi-Fi (To have an established connection you might need to wait an additional second after the Wi-Fi icon appears) 
6. Tap Retry 
7. You should see the previously failed Course/Tab/File row switches state from Error to Loading
8. If you previously successfully downloaded a couple of files, the total number of bytes should decrease because we are only downloading the ones that failed.
 
https://github.com/instructure/canvas-ios/assets/110813321/71f9f931-87a0-45a4-b250-a8ee06ceaa01

## Checklist
